jnius/reflect.py

```diff
@@ -217,58 +232,82 @@ def autoclass(clsname):
+            continue
+        clsDone.add(cls)
+        #as we are walking the entire hierarchy, we only need getDeclaredMethods()
+        #to get what is in this class; other parts of the hierarchy will be found
+        #in those respective classes.
+        methods = cls.getDeclaredMethods()
         methods_name = [x.getName() for x in methods]
-
         for index, method in enumerate(methods):
             name = methods_name[index]
-            if name in classDict:
-                continue
-
-            # only one method available
-            if methods_name.count(name) == 1:
-                static = Modifier.isStatic(method.getModifiers())
-                varargs = method.isVarArgs()
-                sig = '({0}){1}'.format(
-                    ''.join([get_signature(x) for x in method.getParameterTypes()]),
-                    get_signature(method.getReturnType()))
-                if log.level <= logging.DEBUG:
-                    log_method(method, name, sig)
-                classDict[name] = (JavaStaticMethod if static else JavaMethod)(sig, varargs=varargs)
-                if name != 'getClass' and bean_getter(name) and len(method.getParameterTypes()) == 0:
-                    lowername = lower_name(name[2 if name.startswith('is') else 3:])
-                    classDict[lowername] = (lambda n: property(lambda self: getattr(self, n)()))(name)
-                continue
-
+            #print("%s %s" % (str(cls), name))
+
+            clsMethods[name].append((cls, method, level))
+    
+    #having collated the mthods, identify if there are any with the same name
+    for name in clsMethods:        
+        if len(clsMethods[name]) == 1:
+            #uniquely named method
+            owningCls, method, level = clsMethods[name][0]
+            static = Modifier.isStatic(method.getModifiers())
+            varargs = method.isVarArgs()
+            sig = '({0}){1}'.format(
+                ''.join([get_signature(x) for x in method.getParameterTypes()]),
+                get_signature(method.getReturnType()))
+            if log.level <= logging.DEBUG:
+                log_method(method, name, sig)
+            classDict[name] = (JavaStaticMethod if static else JavaMethod)(sig, varargs=varargs)
+            if name != 'getClass' and bean_getter(name) and len(method.getParameterTypes()) == 0:
+                lowername = lower_name(name[2 if name.startswith('is') else 3:])
+                classDict[lowername] = (lambda n: property(lambda self: getattr(self, n)()))(name)
+        else:
             # multiple signatures
             signatures = []
-            for index, subname in enumerate(methods_name):
-                if subname != name:
+            #print("method with %d multiple signatures is %s in cls %s" % (len(clsMethods[name]), name, c))
+            
+            
+            #assume there can be no more than 10000 levels of inheritance
+            paramsig_to_level=defaultdict(lambda: 10000)
```

Comment:
I don't think you need to assume that, you can use `float('inf')` as a starting point :).